clusterloader2/pkg/framework/framework.go

```diff
@@ -167,19 +171,11 @@ func (f *Framework) DeleteAutomanagedNamespaces() *errors.ErrorList {
 
 // DeleteNamespaces deletes the list of namespaces.
 func (f *Framework) DeleteNamespaces(namespaces []string) *errors.ErrorList {
-	var wg wait.Group
+	var wg sync.WaitGroup
 	errList := errors.NewErrorList()
 	for _, namespace := range namespaces {
-		clientSet := f.clientSets.GetClient()
```

Comment:
This method is indeed erroneous, the problem is that you are capturing the loop variable (namespace) in the inline func you pass to the wg.Start. In the next iteration the value of the namespace variable will change and this is causing the race conditions.

This can be fixed with a one liner:
```
for _, namespace := range namespaces {
  namespace := namespace // One line fix, namespace is now not a loop variable anymore
  clientSet  = ...
```

Reply:
cool!